moabb/datasets/base.py

```diff
@@ -248,19 +256,25 @@ def to_int(x):
         except ValueError:
             return x
 
-    values = [str(to_int(val)) for val in row.values]
-    widths = [max(len(col), len(val)) for col, val in zip(col_names, values)]
-    row_sep = " ".join([tab_sep * width for width in widths])
-    cols_row = " ".join([col.rjust(width) for col, width in zip(col_names, widths)])
-    values_row = " ".join([val.rjust(width) for val, width in zip(values, widths)])
-    out = (
-        "    .. admonition:: Dataset summary\n\n"
-        f"{tab_prefix}{row_sep}\n"
-        f"{tab_prefix}{cols_row}\n"
-        f"{tab_prefix}{row_sep}\n"
-        f"{tab_prefix}{values_row}\n"
-        f"{tab_prefix}{row_sep}"
+    keys, values = zip(
+        *[_transfer_unit(str(key), str(to_int(val))) for key, val in row.items()]
     )
+    if horizontal:
+        widths = [max(len(key), len(val)) for key, val in zip(keys, values)]
+        row_sep = " ".join([tab_sep * width for width in widths])
+        keys_str = " ".join([key.rjust(width) for key, width in zip(keys, widths)])
+        values_str = " ".join([val.rjust(width) for val, width in zip(values, widths)])
+        rows = [row_sep, keys_str, row_sep, values_str, row_sep]
+    else:
+        w_keys = max(len(key) for key in keys)
+        w_values = max(len(val) for val in values)
+        row_sep = f"{tab_sep * w_keys} {tab_sep * w_values}"
+        rows = [row_sep]
+        for key, val in zip(keys, values):
+            rows.append(f"{key.rjust(w_keys)} {val.rjust(w_values)}")
+        rows.append(row_sep)
+    rows_str = "\n".join([f"{tab_prefix}{row}" for row in rows])
```

Comment:
Can you just document a little more of this transformation, just small comments on the each line it is good for me :)